examples/recommendations/generate_budget_recommendations.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This example generates budget recommendations on-the-fly for Performance Max campaigns.
+
+Use this example to get budget recommendations during campaign creation workflows.
+
+This example uses the following: 
+1) Performance Max for the campaign type
+2) United States for the geo targeting 
+3) Maximize Conversions Value for the bidding strategy
+
+To get impact metrics for a custom budget, run get_recommendation_impact_metrics.py.
+"""
+
+
+import argparse
+import sys
+
+from google.ads.googleads.client import GoogleAdsClient
+from google.ads.googleads.errors import GoogleAdsException
+
+
+def main(client, customer_id):
+    """The main method that creates all necessary entities for the example.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        customer_id: a client customer ID.
+    """
+    # Start Recommendation Service.
+    recommendation_service = client.get_service("RecommendationService")
+
+    # Build Request.
+    request = client.get_type("GenerateRecommendationsRequest")
+
+    request.customer_id = customer_id
+    request.recommendation_types = ["CAMPAIGN_BUDGET"]
+    request.advertising_channel_type = client.enums.AdvertisingChannelTypeEnum.PERFORMANCE_MAX
+    request.bidding_info.bidding_strategy_type = "MAXIMIZE_CONVERSION_VALUE"
+    request.positive_locations_ids = [2840]  # 2840 is for United States
+    request.asset_group_info = [{ "final_url": "https://www.your-company.com/" }]
+
+    # Send Request.
+    results = recommendation_service.generate_recommendations(request)
+
+    recommendations = results.recommendations
+
+    # Initialize a list to store all budget recommendations with impact metrics.
+    budget_recommendations_list = []
+    # Initialize a list to store budget recommendation amounts.
+    budget_amounts =  []
+
+    # Get budget recommendations with their associated impact metrics.
+    for rec in recommendations:
+        if hasattr(rec, 'campaign_budget_recommendation'):
+            campaign_budget_rec = rec.campaign_budget_recommendation
+            for budget_option in campaign_budget_rec.budget_options:
```

Comment:
It might be helpful to add a comment above this for loop, to explain why we go through different budget options and what we're looking for. E.g. "Loop through the budget options in the campaign budget recommendation to compile a list of budget amounts and their respective potential impact metrics. If you have a campaign creation interface, you could display this information for end users to decide which budget amount best aligns with their goals."